src/Npgsql/NpgsqlReadBuffer.cs

```diff
@@ -160,13 +160,16 @@ async Task EnsureLong()
                 }
 
                 var finalCt = cancellationToken;
-                if (async && Timeout > TimeSpan.Zero)
-                    finalCt = _timeoutCts.Start(cancellationToken);
-
-                try
+                if (async)
+                    finalCt = Timeout > TimeSpan.Zero
+                        ? Cts.Start(cancellationToken)
```

Comment:
So cancellationToken is only interesting when we're called from WaitAsync, right? If so I'll maybe think about another way to handle it (but separately, later).

Reply:
@vonzshik can you confirm that this is your understanding as well?

Reply:
Yep, it is so. Can confirm 😄

Reply:
We've completely forgotten about copy operations, haven't we...

Reply:
I did have them in the back of my mind, though yeah :rofl: 

I've added some thoughts in https://github.com/npgsql/npgsql/issues/1328#issuecomment-713486265, which we can use to track all the COPY work. I still don't see a reason for us to pass the user cancellation token all the way down to the buffer - but I'll hold off on doing any refactoring until we discuss this between us and agree.

Reply:
Also, `NpgsqlTransaction` :trollface: